src/components/app/MenuButtons/MetaOverheadStatistics.js

```diff
@@ -0,0 +1,161 @@
+        // and calculating them.
+        for (const stat of statKeys) {
+          const max = statistics['max' + stat];
+          const mean = statistics['mean' + stat];
+          const min = statistics['min' + stat];
+
+          let currentStat = calculatedStats.get(stat);
+          if (currentStat === undefined) {
+            currentStat = new ProfilerStats();
+            calculatedStats.set(stat, currentStat);
+          }
+
+          currentStat.count(min, max, mean, samplingCount);
+        }
+      }
+    }
+
+    return calculatedStats.size > 0 ? (
+      <details>
+        <summary className="arrowPanelSubTitle">Profiler Overhead</summary>
+        <div className="arrowPanelSection">
+          <div className="metaInfoGrid">
+            <div />
+            <div>Mean</div>
+            <div>Max</div>
+            <div>Min</div>
+            {[...calculatedStats].map(([key, val]) => (
+              <React.Fragment key={key}>
+                <div>{key}</div>
+                <div>{formatNanoseconds(val.mean)}</div>
+                <div>{formatNanoseconds(val.max)}</div>
+                <div>{formatNanoseconds(val.min)}</div>
+              </React.Fragment>
+            ))}
+          </div>
+
+          {overheadDurations !== 0 ? (
+            <div className="metaInfoRow">
+              <span className="metaInfoWideLabel">Overhead Durations:</span>
+              <span className="metaInfoValueRight">
+                {formatNanoseconds(overheadDurations / totalSamplingCount)}
+              </span>
+            </div>
+          ) : null}
+          {overheadPercentage !== 0 ? (
+            <div className="metaInfoRow">
+              <span className="metaInfoWideLabel">Overhead Percentage:</span>
+              <span className="metaInfoValueRight">
+                {formatPercent(overheadPercentage / totalSamplingCount)}
+              </span>
+            </div>
+          ) : null}
+          {profiledDuration !== 0 ? (
+            <div className="metaInfoRow">
+              <span className="metaInfoWideLabel">Profiled Duration:</span>
+              <span className="metaInfoValueRight">
+                {formatNanoseconds(profiledDuration / totalSamplingCount)}
+              </span>
+            </div>
+          ) : null}
+        </div>
+      </details>
+    ) : null;
+  }
+}
+
+// A helper class to calculate the weighted arithmetic mean statistic values.
+// For example let's say that we have 2 processes with these profile stats.
+// Process 1: samplingCount: 90, maxCounter: 10
+// Process 2: samplingCount: 10, maxCounter: 100
+// Since the process 2 has less samples, the weighted average maxCounter going to be
+// (90 * 10 + 10 * 100) / (10 + 90) = 19
+class ProfilerStats {
+  _accumulatedMax = 0;
+  _accumulatedMin = 0;
+  _accumulatedMean = 0;
+  _accumulatedWeight = 0;
+
+  count(min: number, max: number, mean: number, weight: number) {
```

Comment:
nit: I'd use `accumulate` instead of `count`, to keep the same terminology.

When I read `count` the first time, I thought it was doing the final computation; I think `accumulate` is more explicit.